include/ignition/gazebo/Util.hh

```diff
@@ -140,6 +141,14 @@ namespace ignition
         const Entity &_entity,
         const EntityComponentManager &_ecm);
 
+    /// \brief Get all of the links that belong to a model. If the model has
+    /// nested models, the links for the nested models will also be retrieved.
+    /// \param[in] _model The model to retrieve links from.
+    /// \param[in] _ecm The entity component manager.
+    /// \return All of the links that belong to _model.
+    std::unordered_set<Entity> allModelLinks(const Entity &_model,
+        const EntityComponentManager &_ecm);
```

Comment:
What about adding a new method in `Model` that complements the following but works with nested models? It could be a useful helper there, I fear that adding it to `Util.hh` would not get enough visibility and could get lost.

https://github.com/ignitionrobotics/ign-gazebo/blob/5d2c076dfd16886146d62df2ac0bc8e8dea5fdc8/include/ignition/gazebo/Model.hh#L148-L152

Reply:
Thanks for pointing out this method in `Model` - I didn't even think to check this class when working on the PR originally :upside_down_face: so, if I understand correctly, you're recommending that I should move the `allModelLinks` method from `Util.hh` to `Model.hh`? If so, I think that makes sense.

Reply:
Yes, as far as I know, all these classes (`World`, `Model`, `Link`, ...) are abstractions of the ECM. From my point of view, they are useful to 1) act as an intermediate layer detaching the low-level logic that interfaces directly with the ECM so that this logic can be updated / optimized in upstream and directly propagated downstream, and 2) expose high-level functionalities that need getting and processing raw component data (e.g. the [`Link::AddWorldWrench`](https://github.com/ignitionrobotics/ign-gazebo/blob/5d2c076dfd16886146d62df2ac0bc8e8dea5fdc8/include/ignition/gazebo/Link.hh#L245-L253)).

Maybe also this new functionality to get all the links considering a nested topology could be useful to complement the existing `Model::Links`. If you agree with it, maybe moving the function from `Util.hh` to the `Model` class might makes sense.

Reply:
Thanks @diegoferigo and @azeey for your inputs. I made some changes that address https://github.com/ignitionrobotics/ign-gazebo/pull/876#discussion_r660714580 in https://github.com/ignitionrobotics/ign-gazebo/commit/862f0b7dc93b332b8890621e2b4d982fe51688ce, and as a result, I actually don't need the `allModelLinks` method anymore. So, I believe we can mark this as resolved, unless either of you think it would still be worth adding a method like this to the `Model` class (but since this would now be functionality that's unrelated to this PR, perhaps we should open a separate PR to do this if we decide to add it in).

Reply:
If it's not needed here, opening a separate PR sounds good.